server/server.go

```diff
@@ -32,49 +34,51 @@ func StartHttpServer(opts *options.Options) error {
 	return nil
 }
 
-// Check that we can open a TPC connection to all the ports in opts.Ports
-func checkTcpPorts(opts *options.Options) *httpResponse {
+func checkHealthChecks(opts *options.Options) *httpResponse {
 	logger := opts.Logger
 	logger.Infof("Received inbound request. Beginning health checks...")
 
-	allPortsValid := true
-
+	// initialize failedChecks to 0
+	var failedChecks uint64
 	var waitGroup = sync.WaitGroup{}
 
-	for _, port := range opts.Ports {
+	for _, tcpCheck := range opts.Checks.TcpChecks {
```

Comment:
So it looks like this is still doing solely TCP checks?

Reply:
It is - so my plan was to just iterate over each of the different Checks.* - maybe I need to think up a better solution for this one...

Reply:
Thought about this some more - definitely want to switch this to using a common interface - it will then have a method set of (what i can think of so far):

DoCheck (as you mentioned)
ValidateCheck - this would be really handy inside the parseChecksFile to ensure the "required" properties of each check is set... otherwise it currently will error when it's too late (during healthchecks)